### PR TITLE
chore: migrate to gradle version catalogs

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Initial migration to version catalogs (https://docs.gradle.org/current/userguide/platforms.html) 
This change allows us, in next steps (if we want) sharing dependencies across modules, ie: publishing this catalog and conventions (https://docs.gradle.org/current/samples/sample_convention_plugins.html)

Notes:

- Catalogs are supported natively since Gradle 7.4 and integrates with autocompletion and code generation for the IDE without extra configurations
- Now dependencies must be defined in the catalog file `libs.versions.toml`

### Solutions

Removes the class `Dependencies.kt` on `buildSrc` directory and use the convention of a single file `libs.versions.toml`

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by vitorhugods** [comment]
I was considering going it a couple of days ago 😅
Next stop: convention plugins to avoid copy pasting of gradle configurations.

File: logic/build.gradle.kts [hunk 1]
```diff
@@ -55,53 +56,53 @@ kotlin {
...
-                implementation(Dependencies.Coroutines.core)
-                implementation(Dependencies.Kotlinx.serialization)
-                implementation(Dependencies.Kotlinx.dateTime)
-                implementation(Dependencies.UUID.benAsherUUID)
+                implementation(libs.coroutines.core.map {
+                    project.dependencies.create(it, closureOf<ExternalModuleDependency> {
+                        version { strictly(libs.versions.coroutines.get()) }
+                    })
+                })
+                implementation(libs.coroutines.core)
```
Comment (vitorhugods) [inline logic/build.gradle.kts:64]: I noticed recently that we have this duplication of `coroutines-core`.

No need to fix it now, I can remove it when upgrading to Kotlin 1.7.

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌 Just great. Something we needed!

File: logic/build.gradle.kts (hunk 1 above)
Comment (yamilmedina) [reply logic/build.gradle.kts]: Great catch 🦅

**Pull request merged**